NitroxClient/GameLogic/InitialSync/ClockSyncInitialSyncProcessor.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.Communication.NetworkingLayer.LiteNetLib;
+using NitroxClient.GameLogic.InitialSync.Abstract;
+using NitroxClient.GameLogic.Settings;
+using NitroxClient.MonoBehaviours.Gui.Modals;
+using NitroxModel.Networking;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.GameLogic.InitialSync;
+
+public class ClockSyncInitialSyncProcessor : InitialSyncProcessor
+{
+    private readonly TimeManager timeManager;
+    private readonly NtpSyncer ntpSyncer;
+    private readonly LiteNetLibClient liteNetLibClient;
+
+    public ClockSyncInitialSyncProcessor(TimeManager timeManager, NtpSyncer ntpSyncer, IClient client)
+    {
+        this.timeManager = timeManager;
+        this.ntpSyncer = ntpSyncer;
+        liteNetLibClient = (LiteNetLibClient)client;
+
+        AddStep(initialSync => NTPSync(initialSync.TimeData.TimePacket));
+    }
+
+    public IEnumerator NTPSync(TimeChange timeData)
+    {
+        timeManager.SetServerCorrectionData(timeData.OnlineMode, timeData.UtcCorrectionTicks);
+
+        ntpSyncer.Setup(true);
+        ntpSyncer.RequestNtpService();
+
+        yield return new WaitUntil(() => ntpSyncer.Finished);
+
+        if (ntpSyncer.OnlineMode)
+        {
+            timeManager.SetClientCorrectionData(true, ntpSyncer.CorrectionOffset);
+            // If server AND client are in online mode, we have everything we need
+            if (timeData.OnlineMode)
+            {
+                yield break;
+            }
+        }
+
+        Log.Warn($"Both client ({(ntpSyncer.OnlineMode ? "ONLINE" : "OFFLINE")}) and server ({(timeData.OnlineMode ? "ONLINE" : "OFFLINE")}) aren't in ONLINE mode. Falling back to {nameof(ClockSyncProcedure)}");
```

Comment:
The intention of this log feels like they should both be in offline at this point. If that is the case then why are we doing Online/Offline, if one can be online and the other offline here I think we need to reword this warn

Reply:
The issue is that we never know about the actual user's configuration for server/client (there could always be complex networks in use). This system is good enough to be able to treat all cases which is why I don't assume that both clients and server should be offline at that point. This log will happen if either the client or the server isn't in online mode. I'm not sure which part of it needs a reword tho

Reply:
Okay sure if that is the intended then I think we can keep it like this. Rewording it will produce the same result